casper/src/main/scala/coop/rchain/casper/engine/Running.scala

```diff
@@ -298,13 +298,13 @@ class Running[F[_]: Concurrent: BlockStore: CasperBufferStorage: BlockRetriever:
     for {
       // This long chain of ifM created to minimise computation for ignore check and provide
       // readable output for each ignore reason
-      received <- BlockRetriever[F].received(hash)
-      r <- casper.getBlocksInProcessing
-            .map(_.contains(hash))
+      received <- BlockRetriever[F].isReceived(hash)
+      r <- casper.getBlockProcessingState
+            .map(_.processing.contains(hash))
             .ifM(
               IgnoreCasperMessageStatus(doIgnore = true, BlockIsInProcessing).pure[F],
-              BlockRetriever[F].getEnqueuedToCasper
-                .map(_.contains(hash))
+              casper.getBlockProcessingState
```

Comment:
maybe it makes more sense that only call `casper.getBlockProcessingState` once and reused the state.
I am wondering if it is possible that the block is in the `enqueued` first then in `processing` but the double get getBlockProcessingState would skip these situation.

Reply:
@zsluedem nice catch! This makes sense. Though this is not really harmful - purpose of this method if to lower down the unnecessary load. Changes pushed.